cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/AbstractCluster.java

```diff
@@ -168,11 +168,15 @@ protected PersistentVolumeClaim createPersistentVolumeClaim(String name) {
         Map<String, Quantity> requests = new HashMap<>();
         requests.put("storage", storage.size());
 
+        Map<String, String> annotations = new HashMap<>();
+        annotations.put(Storage.DELETE_CLAIM_FIELD, String.valueOf(storage.isDeleteClaim()));
```

Comment:
Smart use of annotation. We should not forget to update the annotation in the update PR. 

Only one comment. I think the practice is to prefix the annotation with a domain name - e.g. `cluster-contorller.stirmzi,io/dalateClaim`.

Reply:
What do you mean with "update the annotation in the update PR" ? Sorry .. :-(

> Only one comment. I think the practice is to prefix the annotation with a domain name - e.g. cluster-contorller.stirmzi,io/dalateClaim.

Totally agree, I started trying to use the already defined constant with "delete-claim" value avoiding to add another string but what you are suggesting is the right way to go.

Reply:
> What do you mean with "update the annotation in the update PR" ? Sorry .. :-(

When the cluster config map is updated and perhaps the deleteClaim flag changes we need to make sure to change the annotation. We should not set it only when creating the SattefulSet.

Reply:
Right. Got it.